### PR TITLE
formatjson5: init at 0.2.6

### PR DESCRIPTION
**Pull request by katexochen**
## Description of changes

A JSON5 formatter

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fo/formatjson5/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+rustPlatform.buildRustPackage rec {
+  pname = "formatjson5";
+  version = "0.2.6";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "json5format";
+    # Not tagged, see Cargo.toml.
+    rev = "056829990bab4ddc78c65a0b45215708c91b8628";
+    hash = "sha256-35ftRr0XEU1AC98MiP8Q6dWIChPHcATkqPYfoVzZRmY=";
```
Comment (AndersonTorres) [inline pkgs/by-name/fo/formatjson5/package.nix:15]: I have tried to build it via nixpkgs-review, but it failed with  hash mismatch:

```
error: hash mismatch in fixed-output derivation '/nix/store/p35c72i9bk1v98q5yra7i2hy5vp06akn-source.drv':
         specified: sha256-35ftRr0XEU1AC98MiP8Q6dWIChPHcATkqPYfoVzZRmY=
            got:    sha256-Lredw/Fez+2U2++ShZcKTFCv8Qpai9YUvqvpGjG5W0o=
error: 1 dependencies of derivation '/nix/store/wcnizz1i4jq8jhybyj1afzw0p3lf4kw0-formatjson5-0.2.6-vendor.tar.gz.drv' failed to build
error: 2 dependencies of derivation '/nix/store/j2hx1g8mai2b747pbdrsikmqrn0dlifr-formatjson5-0.2.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a9kipv2w1n00p7m0jzs4vi3jzg7lcgch-review-shell.drv' failed to build
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/fo/formatjson5/package.nix (hunk 1 above)
Comment (katexochen) [reply pkgs/by-name/fo/formatjson5/package.nix]: Thanks, should be fixed now. I moved the package to another dir and the build was still cached.

**Comment by malt3** [comment]
Result of `nixpkgs-review pr 278690` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>formatjson5</li>
  </ul>
</details>

**Review by malt3 — APPROVED** [review]
Binary works on Linux amd64.

**Review by msanft — APPROVED** [review]
Built and tested basic functionality on `x86_64-linux`

**Review by drupol — CHANGES_REQUESTED** [review]
It doesn't compile on Darwin ? I think you're missing dependencies for Darwin.
If you do not run Darwin, it's not so hard to add, maybe something like that should do it:

```
  nativeBuildInputs = [ ... ] ++ lib.optionals (stdenv.isDarwin) [ darwin.apple_sdk.frameworks.Security ];
```

There's quite a few examples in `nixpkgs`.

**Comment by drupol** [comment]
@GrahamcOfBorg build formatjson5

**Review by drupol — APPROVED** [review]
The CI issue is something global: https://github.com/NixOS/nixpkgs/pull/281363

For the rest, this can be merged once the last darwin build succeed.

**Comment by drupol** [comment]
Merging this, waiting more than 10 hours for a simple build is simply not sustainable.

**Pull request merged**